lib/compute/index.js

```diff
@@ -0,0 +1,600 @@
+ *   }, callback);
+ */
+Compute.prototype.createFirewall = function(name, options, callback) {
+  if (!name) {
+    throw new Error('A firewall name is needed to use Google Cloud Compute.');
+  } else if (!/^(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)$/.test(name)) {
+    throw new Error('Name must match [a-z]([-a-z0-9]{0,61}[a-z0-9])?');
+  }
+  if (!options || !util.is(options.allowed, 'array')) {
+    throw new Error('Allowed protocols and ports must be provided.');
+  }
+  if (!util.is(options.sourceRanges, 'array') && 
+    !util.is(options.sourceTags, 'array')) {
+    throw new Error('Source ranges or source tags must be provided.');
+  }
+
+  var body = {};
+  body.name = name;
+  if (util.is(options.description, 'string')) {
+    body.description = options.description;
+  }
+  if (util.is(options.network, 'string')) {
+    body.network = options.network;
+  } else {
+    body.network = 'global/networks/default';
+  }
+  if (util.is(options.allowed, 'array')) {
+    body.allowed = options.allowed;
+  } 
+  if (util.is(options.sourceRanges, 'array')) {
+    body.sourceRanges = options.sourceRanges;
+  } 
+  if (util.is(options.sourceTags, 'array')) {
+    body.sourceTags = options.sourceTags;
+  } 
+  if (util.is(options.targetTags, 'array')) {
+    body.targetTags = options.targetTags;
+  } 
+
+  this.makeReq_(
+    'POST', 
+    '/global/firewalls', 
+    null, 
+    body, function(err, resp) {
+      if (err) {
+        callback(err);
+        return; 
+      }
+      var firewall = new Firewall(this, name, body);
+      callback(null, firewall, resp);
+    }.bind(this)); 
+};
+
+/**
+ * Get a list of instances.
+ *
+ * @throws {Error} if a malformed filter is provided.
+ *
+ * @param {object} options - Instance search options.
+ * @param {number=} options.maxResults - Maximum number of instances to 
+ *     return.
+ * @param {string=} options.zone - Only instances in this zone a returned.
+ * @param {object=} options.filter - Search filter.
+ * @param {string} options.filter.fieldName - Instance field to consider in this
+ *     filter.
+ * @param {string} options.filter.operatorString - Filter operator, can be
+ *     either 'eq' or 'ne'.
+ * @param {string} options.filter.literalString - String to compare the instance
+ *     field to. Can be a regular expression.
+ * @param {string=} pageToken - Page identifier used in paginated search.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var callback = function(err, instances) {
+ *   // `instances` is an array of `Instance` objects.
+ * };
+ *
+ * var compute = gcloud.compute({
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * compute.listInstances( 
+ *   {
+ *     zone: 'europe-west1-b',
+ *     filter: {
+ *       fieldName : 'name',
+ *       operatorString : 'eq',
+ *       literalString : 'instance-[0-9]'
+ *     }],
+ *   }, callback);
+ */
+Compute.prototype.listInstances = function(options, callback) {
+  if (!callback) {
+    callback = options;
```

Comment:
I think rather than assuming options is meant to be the callback, we could do something like

```javascript
if (util.is(options, 'function')) { ... }
```

And at the same time we can set a default value for options and skip the check on line 318 altogether.